numpy/_core/tests/test_multiarray.py

```diff
@@ -8501,10 +8502,57 @@ def __array__(self, dtype=None):
                     match=r"Unable to avoid copy(.|\n)*numpy_2_0_migration_guide.html"):
                 np.array(a, copy=False)
 
+    @pytest.mark.skipif(IS_PYPY, reason="PyPy copies differently")
+    @pytest.mark.skipif(not HAS_PSUTIL, reason="psutil isn't installed")
+    def test___array__copy_once(self):
+        import psutil
+
+        size = 1000
+        base_arr = np.zeros((size, size))
+
+        class ArrayRandom:
+            def __array__(self, dtype=None, copy=None):
+                if copy:
+                    return np.random.randn(size, size)
+                else:
+                    return base_arr
+
+        process = psutil.Process()
```

Comment:
Can you just do this without `psutils`:  Just store *copy* you return, then compare do an identity/base check.  Easier, and no need for psutil magic.

Reply:
Ah, right - I removed `psutils` and stored copies/base arrays instead.